utils/clean/clean.go

```diff
@@ -0,0 +1,64 @@
+// pmm-managed
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+// Package clean has the old actions results cleaner.
+package clean
+
+import (
+	"context"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"gopkg.in/reform.v1"
+
+	"github.com/percona/pmm-managed/models"
+)
+
+// CleanResults has unexported fields for the results cleanup function.
+type CleanResults struct {
+	db *reform.DB
+}
+
+// New returns a new CleanResults instance.
+func New(db *reform.DB) *CleanResults {
+	return &CleanResults{db: db}
+}
+
+// Run starts the clean process.
+func (c *CleanResults) Run(ctx context.Context, interval time.Duration) {
+	ticker := time.NewTicker(interval)
+	l := logrus.WithField("component", "cleaner")
+
+	for {
+		// if this query runs every minute, it will clean results older than 1 minute,
```

Comment:
Maybe it's better to have 2 separate variables for them?

Reply:
It's not addressed nor answered yet

Reply:
If we add a second parameter to change the deletion range we would end with 2 scenarios:
**First scenario:** run every 2 minutes, delete results older than 4 minutes
```
00:00 x deleted on first run
00:01 x deleted on first run
00:02 x deleted on first run
00:03 x deleted on second run
00:04 x deleted on second run
00:05
00:06 First run <- ticker starts here
00:07
00:08 Second run
00:09
00:10
00:11
00:12
00:13
00:14
00:15
```
**Second scenario:** run every 4 minutes, delete results older than 2 minutes
```
00:00 x deleted on first run
00:01 x deleted on first run
00:02 x deleted on first run
00:03 x deleted on first run
00:04 x deleted on second run
00:05 x deleted on second run
00:06 x deleted on second run <-- First run - ticker starts here
00:07 x deleted on second run
00:08 
00:09
00:10 Second run
00:11
00:12
00:13
00:14
00:15
```

**Current implementation:** run every 2 minutes, delete results older than 2 minutes
```
00:00 x deleted on first run
00:01 x deleted on first run
00:02 x deleted on first run
00:03 x deleted on first run
00:04 x deleted on second run
00:05 x deleted on second run
00:06 <-- First run - ticker starts here
00:07 
00:08 <- Second run 
00:09
00:10 
00:11
00:12
00:13
00:14
00:15
```
The current implementation keeps results clean enough and it is easier to understand how it works. 
If we add another parameter, I would also add a validation to ensure we are in the first scenario.

Reply:
@AlekSi what do you think?

Reply:
I think we will run it every 1 hour and remove everything older than 24 hours.

Reply:
1 hour / 24 hours values are too high.

I think the sliding window approach (run every X minutes, delete rows older than Y minutes) is better overall for storage consumption.

I also think X and Y should be named constants until we really have a usecase for them being configurable.